src/librbd/cache/ObjectCacherObjectDispatch.cc

```diff
@@ -128,7 +128,7 @@ void ObjectCacherObjectDispatch<I>::init() {
                 << " max_dirty_age=" << max_dirty_age << dendl;
 
   m_object_cacher = new ObjectCacher(cct, m_image_ctx->perfcounter->get_name(),
-                                     *m_writeback_handler, m_cache_lock,
```

Comment:
Nit: `ObjectCacheObjectDispatch` and `ObjectCacherWriteback` have assumptions about `m_cache_lock`. Now while I am fine with removing it, these assumptions in `librbd` need to be fixed. Preferably by removing `ImageCtx::m_cache_lock` and adding a new lock to `ObjectCacherWriteback` for protecting the data structures.

Reply:
Sure, will check it.

Reply:
@dillaman 
Done, have made the m_lock a private one in `ObjectCacherWriteback`.

Reply:
And the `./bin/ceph_test_librbd` test has succeeded. Is that enough ?